.github/workflows/ci.yml

```diff
@@ -18,6 +19,15 @@ jobs:
         # https://github.com/joaomcteixeira/python-project-skeleton/releases/tag/v0.10.1
         # below the list of tox envs that you wish to include in the matrix
         tox-envs: [docs, lint, build, test, pr]
+        # workaround to ignore 'pr' when merging to main
+        # see: https://github.community/t/how-to-conditionally-include-exclude-items-in-matrix-eg-based-on-branch/16853/6
+        isMerge:
+            - ${{ github.event_name == 'push' && github.ref == 'refs/heads/main' }}
```

Comment:
For anyone reaching this PR trying to make the exclusion work with a boolean input, called useBeta for example, (workflow_dispatch) I had to use `${{ inputs.useBeta }}` instead of `${{ github.event.inputs.useBeta }}`. I think it is because `inputs` maintains the boolean whilst `github.events.inputs` converts it to string: https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#providing-inputs

By the way, very useful example here, thanks!